x-pack/plugins/enterprise_search/public/applications/shared/flash_messages/flash_messages_logic.test.ts

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { resetContext } from 'kea';
+
+import { FlashMessagesLogic, IFlashMessage } from './flash_messages_logic';
+
+describe('FlashMessagesLogic', () => {
+  const DEFAULT_VALUES = {
+    messages: [],
+    queuedMessages: [],
+    historyListener: null,
+  };
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+    resetContext({});
+  });
+
+  it('has expected default values', () => {
+    FlashMessagesLogic.mount();
+    expect(FlashMessagesLogic.values).toEqual(DEFAULT_VALUES);
+  });
+
+  describe('setFlashMessages()', () => {
+    it('sets an array of messages', () => {
+      const messages = [
+        { type: 'success', message: 'Hello world!!' },
+        { type: 'error', message: 'Whoa nelly!', description: 'Uh oh' },
+        { type: 'info', message: 'Everything is fine, nothing is ruined' },
+      ] as IFlashMessage[];
+
+      FlashMessagesLogic.mount();
+      FlashMessagesLogic.actions.setFlashMessages(messages);
+
+      expect(FlashMessagesLogic.values.messages).toEqual(messages);
+    });
+
+    it('automatically converts to an array if a single message obj is passed in', () => {
+      const message = { type: 'success', message: 'I turn into an array!' } as IFlashMessage;
+
+      FlashMessagesLogic.mount();
+      FlashMessagesLogic.actions.setFlashMessages(message);
+
+      expect(FlashMessagesLogic.values.messages).toEqual([message]);
+    });
+  });
+
+  describe('clearFlashMessages()', () => {
+    it('sets messages back to an empty array', () => {
+      FlashMessagesLogic.mount();
+      FlashMessagesLogic.actions.setFlashMessages('test' as any);
+      FlashMessagesLogic.actions.clearFlashMessages();
+
+      expect(FlashMessagesLogic.values.messages).toEqual([]);
+    });
+  });
+
+  describe('setQueuedMessages()', () => {
+    it('sets an array of messages', () => {
+      const queuedMessage = { type: 'error', message: 'You deleted a thing' } as IFlashMessage;
```

Comment:
```suggestion
      const queuedMessage: IFlashMessage = { type: 'error', message: 'You deleted a thing' };
```

Same thing as above RE strictness.  I believe `as` should only be used when passing something as an argument, if you're assigning the value to a variable, you should explicitly declare the type on the variable.